### ISSUE TITLE
Enable debug mode in SimulatedBackend

### ISSUE THREAD

**Issue by chawin-a** [ethereum/go-ethereum#25958]
# Rationale

I attempted to develop test cases for my Go projects. To test locally, I use this [simulator](https://github.com/ethereum/go-ethereum/blob/master/accounts/abi/bind/backends/simulated.go). However, some mathematical calculations in Go and smart contracts differ. I'm not sure why. Only the smart contract's output can be tested using the simulator. I need to go over the code line by line.

# Implementation

I've read the code that the simulator uses to call contracts. This [line](https://github.com/ethereum/go-ethereum/blob/564751668aa01f55d39851cbf4673500ec2d4718/accounts/abi/bind/backends/simulated.go#L651) can enable debugging and create its own tracer to capture optcode, call frame, transaction, and so on. 
This is beneficial to Go developers who are interested in Ethereum.

Issue #25072 is very similar to mine.

Can we write a function SetVMConfig and use it after creating a simulator like this?
```
sim := backends.NewSimulatedBackend(core.GenesisAlloc{}, 10000000)
sim.SetVMConfig(vm.Config{
    Debug: true,
    Tracer: MyTracer,
})
```
or
```
sim := backends.NewSimulatedBackend(core.GenesisAlloc{}, 10000000, vm.Config{
    Debug: true,
    Tracer: MyTracer,
})
```
and change this [line](https://github.com/ethereum/go-ethereum/blob/564751668aa01f55d39851cbf4673500ec2d4718/accounts/abi/bind/backends/simulated.go#L651)
```
vmEnv := vm.NewEVM(evmContext, txContext, stateDB, b.config, vm.Config{NoBaseFee: true})
```
to
```
vmEnv := vm.NewEVM(evmContext, txContext, stateDB, b.config, vm.Config{
    Debug:     b.blockchain.GetVMConfig().Debug,
    Tracer:    b.blockchain.GetVMConfig().Tracer,
    NoBaseFee: true,
})
```

**Comment by MariusVanDerWijden** [comment]
So merging https://github.com/ethereum/go-ethereum/pull/25072 would resolve your problems?

**Comment by aschlosberg** [comment]
I've updated #25072 to be up to date with `master`. I'm also keen to see it merged as it will unblock usage of the [solcover](https://pkg.go.dev/github.com/divergencetech/ethier/solcover) package.

**Comment by chawin-a** [comment]
Yes, merging #25072 would solve our problems.

**Comment by MariusVanDerWijden** [comment]
We have reworked our simulated backend, s.th. it mirrors the normal blockchain much better. Please open a new issue if you still feel like this is an issue
